### PR TITLE
June 28th Upstream Merge 

### PR DESCRIPTION
**Pull request by Jolly-66**
Uhhhh don't tell the nonexistent skrell players I accidentally nuked their define, I think we should remove skrells someday tbh, no one.. uses them. Idk. Define was moved so I guess no harm done, but there may be bugs or whatever. Vulpines when?

Game compiled fine, got no errors, modular content should be fine unless I was stupid dumb piss baby.

Merges code up to ~7pm EST on 6/28

### CODE REVIEW

File: code/__DEFINES/is_helpers.dm [hunk 1]
```diff
@@ -72,7 +75,7 @@ GLOBAL_LIST_INIT(turfs_without_ground, typecacheof(list(
 #define isvampire(A) (is_species(A,/datum/species/vampire))
 #define isdullahan(A) (is_species(A, /datum/species/dullahan))
 #define ismonkey(A) (is_species(A, /datum/species/monkey))
-#define isSkrell(A) (is_species(A, /datum/species/skrell)) //JollyStation Addition, adding is-check for Skrell!
+#define isandroid(A) (is_species(A, /datum/species/android))
```
Comment (MrMelbert) [inline code/__DEFINES/is_helpers.dm:78]: e?

File: code/__DEFINES/mobs.dm [hunk 2]
```diff
@@ -57,7 +57,7 @@
 #define BODYPART_ORGANIC   1
 #define BODYPART_ROBOTIC   2
 
-#define DEFAULT_BODYPART_ICON_ORGANIC 'jollystation_modules/icons/mob/human_parts_greyscale_js.dmi' //Changing from the "icons/mob/human_parts_greyscale" file to the "jollystation_modules/icons/mob/human_parts_greyscale_js" version, to account for Skrell additions. JollyStation Edit -Jon
+#define DEFAULT_BODYPART_ICON_ORGANIC 'icons/mob/human_parts_greyscale.dmi'
```
Comment (MrMelbert) [inline code/__DEFINES/mobs.dm:60]: e?

File: code/__HELPERS/global_lists.dm [hunk 3]
```diff
@@ -34,7 +34,6 @@
 	init_sprite_accessory_subtypes(/datum/sprite_accessory/moth_antennae, GLOB.moth_antennae_list)
 	init_sprite_accessory_subtypes(/datum/sprite_accessory/moth_markings, GLOB.moth_markings_list)
 	init_sprite_accessory_subtypes(/datum/sprite_accessory/tails/monkey, GLOB.tails_list_monkey)
-	init_sprite_accessory_subtypes(/datum/sprite_accessory/skrell_headtentacles, GLOB.skrellheadtentacles_list) //Adding Skrell Headtentacles init -Jon
```
Comment (MrMelbert) [inline code/__HELPERS/global_lists.dm:37]: e?

File: code/__HELPERS/icons.dm [hunk 4]
```diff
@@ -1057,8 +1057,8 @@ GLOBAL_LIST_EMPTY(friendly_animal_types)
 		if(J)
 			J.equip(body, TRUE, FALSE, outfit_override = outfit_override)
 		else if (outfit_override)
-			//body.equipOutfit(outfit_override,visualsOnly = TRUE)
-			body.equip_outfit_and_loadout(outfit_override, TRUE, prefs) // NON-MODULE CHANGE
+			body.equipOutfit(outfit_override,visualsOnly = TRUE)
```
Comment (MrMelbert) [inline code/__HELPERS/icons.dm:1060]: e?

**Comment by Jolly-66** [comment]
image
image
https://www.youtube.com/watch?v=czTksCF6X8Y&t=4s&ab_channel=NickCagnetti

File: code/datums/datacore.dm [hunk 5]
```diff
@@ -272,9 +272,6 @@
 			G.fields["gender"]  = "Other"
 		G.fields["photo_front"] = photo_front
 		G.fields["photo_side"] = photo_side
-		//NON-MODULE CHANGES:
-		G.fields["past_records"] = C.prefs.general_records
-		//NON-MODULE CHANGES END
```
Comment (MrMelbert) [inline code/datums/datacore.dm:277]: e?

File: code/datums/datacore.dm [hunk 6]
```diff
@@ -291,9 +288,6 @@
 		M.fields["cdi_d"] = "No diseases have been diagnosed at the moment."
 		M.fields["notes"] = H.get_quirk_string(!medical, CAT_QUIRK_NOTES)
 		M.fields["notes_d"] = H.get_quirk_string(medical, CAT_QUIRK_NOTES)
-		//NON-MODULE CHANGES:
-		M.fields["past_records"] = C.prefs.medical_records
-		//NON-MODULE CHANGES END
```
Comment (MrMelbert) [inline code/datums/datacore.dm:296]: e?

File: code/datums/datacore.dm [hunk 7]
```diff
@@ -304,9 +298,6 @@
 		S.fields["citation"] = list()
 		S.fields["crim"] = list()
 		S.fields["notes"] = "No notes."
-		//NON-MODULE CHANGES:
-		S.fields["past_records"] = C.prefs.security_records
-		//NON-MODULE CHANGES END
```
Comment (MrMelbert) [inline code/datums/datacore.dm:309]: e?

File: code/modules/client/preferences_savefile.dm [hunk 8]
```diff
@@ -225,10 +226,6 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 			parsed_favs += path
 	favorite_outfits = uniqueList(parsed_favs)
 
-	// NON-MODULE CHANGES: client prefs
-	READ_FILE(S["hear_speech_sounds"] , hear_speech_sounds)
-	READ_FILE(S["hear_radio_sounds"] , hear_radio_sounds)
-
```
Comment (MrMelbert) [inline code/modules/client/preferences_savefile.dm:231]: e

File: code/modules/client/preferences_savefile.dm [hunk 9]
```diff
@@ -278,10 +275,6 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 	key_bindings = sanitize_keybindings(key_bindings)
 	favorite_outfits = SANITIZE_LIST(favorite_outfits)
 
-	// NON-MODULE CHANGES: sanitization
-	hear_speech_sounds = sanitize_integer(hear_speech_sounds, FALSE, TRUE, initial(hear_speech_sounds))
-	hear_radio_sounds = sanitize_integer(hear_speech_sounds, FALSE, TRUE, initial(hear_radio_sounds))
-
```
Comment (MrMelbert) [inline code/modules/client/preferences_savefile.dm:284]: e

File: code/modules/client/preferences_savefile.dm [hunk 10]
```diff
@@ -359,11 +352,6 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 	WRITE_FILE(S["key_bindings"], key_bindings)
 	WRITE_FILE(S["hearted_until"], (hearted_until > world.realtime ? hearted_until : null))
 	WRITE_FILE(S["favorite_outfits"], favorite_outfits)
-
-	// NON-MODULE CHANGES: client prefs
-	WRITE_FILE(S["hear_speech_sounds"] , hear_speech_sounds)
-	WRITE_FILE(S["hear_radio_sounds"] , hear_radio_sounds)
-
```
Comment (MrMelbert) [inline code/modules/client/preferences_savefile.dm:366]: e

File: code/modules/client/preferences_savefile.dm [hunk 11]
```diff
@@ -430,18 +418,6 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
...
-	READ_FILE(S["runechat_color"] , runechat_color)
-	READ_FILE(S["flavor_text"] , flavor_text)
-	READ_FILE(S["security_records"] , security_records)
-	READ_FILE(S["medical_records"] , medical_records)
-	READ_FILE(S["general_records"] , general_records)
-	READ_FILE(S["exploitable_info"] , exploitable_info)
-	READ_FILE(S["loadout_list"] , loadout_list)
-	READ_FILE(S["greyscale_loadout_list"] , greyscale_loadout_list)
-	//NON-MODULE CHANGES END
-
```
Comment (MrMelbert) [inline code/modules/client/preferences_savefile.dm:444]: big e

File: code/modules/client/preferences_savefile.dm [hunk 12]
```diff
@@ -529,20 +505,6 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
...
-	flavor_text = strip_html_simple(sanitize_text(flavor_text), MAX_MESSAGE_LEN)
-	security_records = strip_html_simple(sanitize_text(security_records), MAX_FLAVOR_LEN)
-	medical_records = strip_html_simple(sanitize_text(medical_records), MAX_FLAVOR_LEN)
-	general_records = strip_html_simple(sanitize_text(general_records), MAX_FLAVOR_LEN)
-	exploitable_info = strip_html_simple(sanitize_text(exploitable_info), MAX_FLAVOR_LEN)
-
-	sanitize_loadout_list(loadout_list)
-	sanitize_greyscale_list(greyscale_loadout_list)
-
-	//NON-MODULE CHANGES END
```
Comment (MrMelbert) [inline code/modules/client/preferences_savefile.dm:545]: e

File: code/modules/client/preferences_savefile.dm [hunk 13]
```diff
@@ -604,17 +566,6 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
...
-	WRITE_FILE(S["feature_skrell_headtentacles"], features["skrell_headtentacles"])
-	WRITE_FILE(S["runechat_color"] , runechat_color)
-	WRITE_FILE(S["flavor_text"] , flavor_text)
-	WRITE_FILE(S["general_records"] , general_records)
-	WRITE_FILE(S["security_records"] , security_records)
-	WRITE_FILE(S["medical_records"] , medical_records)
-	WRITE_FILE(S["exploitable_info"] , exploitable_info)
-	WRITE_FILE(S["loadout_list"], loadout_list)
-	WRITE_FILE(S["greyscale_loadout_list"], greyscale_loadout_list)
-	//NON-MODULE CHANGES END
```
Comment (MrMelbert) [inline code/modules/client/preferences_savefile.dm:617]: e

File: code/modules/client/preferences.dm [hunk 14]
```diff
@@ -80,7 +80,7 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 	var/skin_tone = "caucasian1" //Skin color
 	var/eye_color = "000" //Eye color
 	var/datum/species/pref_species = new /datum/species/human() //Mutant race
-	var/list/features = list("mcolor" = "FFF", "ethcolor" = "9c3030", "tail_lizard" = "Smooth", "tail_human" = "None", "snout" = "Round", "horns" = "None", "ears" = "None", "wings" = "None", "frills" = "None", "spines" = "None", "body_markings" = "None", "legs" = "Normal Legs", "moth_wings" = "Plain", "moth_antennae" = "Plain", "moth_markings" = "None", "skrell_headtentacles" = "Male") // NON-MODULE EDIT
```
Comment (MrMelbert) [inline code/modules/client/preferences.dm:83]: e

File: code/modules/client/preferences.dm [hunk 15]
```diff
@@ -5,7 +5,7 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 	//doohickeys for savefiles
 	var/path
 	var/default_slot = 1 //Holder so it doesn't default to slot 1, rather the last one used
-	var/max_save_slots = 6 //NON-MODULE CHANGE
```
Comment (MrMelbert) [inline code/modules/client/preferences.dm:8]: e

File: code/modules/client/preferences.dm [hunk 16]
```diff
@@ -189,13 +189,10 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 	update_preview_icon()
 	var/list/dat = list("<center>")
 
-//NON-MODULE CHANGE:
 	dat += "<a href='?_src_=prefs;preference=tab;tab=0' [current_tab == 0 ? "class='linkOn'" : ""]>Character Settings</a>"
-	dat += "<a href='?_src_=prefs;preference=tab;tab=1' [current_tab == 1 ? "class='linkOn'" : ""]>Character Background</a>"
-	dat += "<a href='?_src_=prefs;preference=tab;tab=2' [current_tab == 2 ? "class='linkOn'" : ""]>Game Preferences</a>"
-	dat += "<a href='?_src_=prefs;preference=tab;tab=3' [current_tab == 3 ? "class='linkOn'" : ""]>OOC Preferences</a>"
-	dat += "<a href='?_src_=prefs;preference=tab;tab=4' [current_tab == 4 ? "class='linkOn'" : ""]>Custom Keybindings</a>"
-//NON-MODULE CHANGE END
```
Comment (MrMelbert) [inline code/modules/client/preferences.dm:198]: e

File: code/modules/client/preferences.dm [hunk 17]
```diff
@@ -547,21 +544,6 @@ GLOBAL_LIST_EMPTY(preferences_datums)
...
-				dat += "<h3>Head Tentacles</h3>"
-
-				dat += "<a href='?_src_=prefs;preference=skrell_headtentacles;task=input'>[features["skrell_headtentacles"]]</a><BR>"
-
-				mutant_category++
-				if(mutant_category >= MAX_MUTANT_ROWS)
-					dat += "</td>"
-					mutant_category = 0
-			//JollyStation Addition End
-
```
Comment (MrMelbert) [inline code/modules/client/preferences.dm:564]: e

File: code/modules/client/preferences.dm [hunk 18]
```diff
@@ -587,54 +569,9 @@ GLOBAL_LIST_EMPTY(preferences_datums)
...
-			dat += "<BR>"
-			dat += 	"<a href='?_src_=prefs;preference=medical_records;task=input'><b>Medical</b></a><br>"
-			dat += "[TextPreview(medical_records, 125)]"
-			dat += "<BR>"
-			dat += 	"<a href='?_src_=prefs;preference=exploitable_info;task=input'><b>Exploitable Information</b></a><br>"
-			dat += "[TextPreview(exploitable_info, 125)]"
-			dat += "<BR><BR>"
-			dat += "</td></tr></table>"
-			//NON-MODULE CHANGES END
-		if (2) // Game Preferences, NON-MODULE CHANGES
```
Comment (MrMelbert) [inline code/modules/client/preferences.dm:637]: huge e

File: code/modules/client/preferences.dm [hunk 19]
```diff
@@ -772,7 +709,7 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 			dat += "<br>"
 			dat += "<b>Midround Antagonist:</b> <a href='?_src_=prefs;preference=allow_midround_antag'>[(toggles & MIDROUND_ANTAG) ? "Enabled" : "Disabled"]</a><br>"
 			dat += "</td></tr></table>"
-		if(3) //OOC Preferences, NON-MODULE CHANGES
```
Comment (MrMelbert) [inline code/modules/client/preferences.dm:775]: e

File: code/modules/client/preferences.dm [hunk 20]
```diff
@@ -851,7 +788,7 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 
 				dat += "</td>"
 			dat += "</tr></table>"
-		if(4) // Custom keybindings, NON-MODULE CHANGES
```
Comment (MrMelbert) [inline code/modules/client/preferences.dm:854]: e

File: code/modules/client/preferences.dm [hunk 21]
```diff
@@ -1365,50 +1302,6 @@ GLOBAL_LIST_EMPTY(preferences_datums)
...
-					var/med_record = input(usr, "Set your medical records", "Medical Records", medical_records) as message|null
-					if(med_record)
-						medical_records = strip_html_simple(med_record, MAX_FLAVOR_LEN)
-
-				if("exploitable_info")
-					var/expl_info = input(usr, "Set your exploitable information, this rarely will be showed to antagonists", "Exploitable Info", exploitable_info) as message|null
-					if(expl_info)
-						exploitable_info = strip_html_simple(expl_info, MAX_FLAVOR_LEN)
-				//NON-MODULE CHANGES END
-
```
Comment (MrMelbert) [inline code/modules/client/preferences.dm:1411]: huge e

File: code/modules/client/preferences.dm [hunk 22]
```diff
@@ -1576,14 +1469,6 @@ GLOBAL_LIST_EMPTY(preferences_datums)
...
 						features["moth_markings"] = new_moth_markings
 
-				//JollyStation Addition Start
-				if("skrell_headtentacles")
-					var/new_skrell_headtentacles
-					new_skrell_headtentacles = input(user, "Choose your character's style of head tentacles:", "Character Preference") as null|anything in GLOB.skrellheadtentacles_list
-					if(new_skrell_headtentacles)
-						features["skrell_headtentacles"] = new_skrell_headtentacles
-				//JollyStation Addition End
-
```
Comment (MrMelbert) [inline code/modules/client/preferences.dm:1586]: e

File: code/modules/client/preferences.dm [hunk 23]
```diff
@@ -1985,12 +1870,6 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 					to_chat(user, span_notice("OOC Commendation Heart disabled"))
 					save_preferences()
 
-				if("hear_speech_sounds")
-					hear_speech_sounds = !hear_speech_sounds
-
-				if("hear_radio_sounds")
-					hear_radio_sounds = !hear_radio_sounds
-
```
Comment (MrMelbert) [inline code/modules/client/preferences.dm:1993]: e

**Review by MrMelbert — CHANGES_REQUESTED** [review]
You've dropped a ton of modular changes

**Comment by Jolly-66** [comment]
fuck this im redoing this another time

**Pull request closed without merging**